crates/astria-sequencer-relayer/src/relayer.rs

```diff
@@ -359,11 +373,11 @@ impl Relayer {
             // layer if no submission is in flight.
             if !self.queued_blocks.is_empty() && self.submission_task.is_none() {
                 let client = self.data_availability.clone();
+                let queued_blocks = std::mem::replace(&mut self.queued_blocks, vec![]);
                 self.submission_task = Some(task::spawn(submit_blocks_to_celestia(
                     client,
-                    self.queued_blocks.clone(),
+                    queued_blocks,
```

Comment:
If there are issues with endpoint this will effectively be a hot loop that:

1. spawns the task
2. immediately fails
3. respawns

Depending on the setup of the other futures in the select! statement it might be that this drowns out all other branches.